src/pytorch_lightning/tuner/batch_size_scaling.py

```diff
@@ -69,61 +53,95 @@ def scale_batch_size(
 
     __scale_batch_restore_params(trainer, params)
 
     if trainer.progress_bar_callback:
         trainer.progress_bar_callback.enable()
 
-    # Restore initial state of model
     trainer._checkpoint_connector.restore(ckpt_path)
     trainer.strategy.remove_checkpoint(ckpt_path)
 
     return new_size
 
 
 def __scale_batch_dump_params(trainer: "pl.Trainer") -> Dict[str, Any]:
-    return {
-        "max_steps": trainer.fit_loop.max_steps,
-        "logger": trainer.logger,
+    dumped_params = {
+        "loggers": trainer.loggers,
         "callbacks": trainer.callbacks,
-        "auto_scale_batch_size": trainer.auto_scale_batch_size,
-        "auto_lr_find": trainer.auto_lr_find,
-        "limit_train_batches": trainer.limit_train_batches,
     }
+    if trainer.state.fn == "fit":
+        loop = trainer.fit_loop
+        dumped_params["max_steps"] = trainer.max_steps
+        dumped_params["limit_train_batches"] = trainer.limit_train_batches
+        dumped_params["limit_val_batches"] = trainer.limit_val_batches
+    else:
+        stage = trainer.state.stage
+        loop = getattr(trainer, f"{stage}_loop")
+        assert stage is not None
+        dumped_params["limit_eval_batches"] = getattr(trainer, f"limit_{stage.dataloader_prefix}_batches")
+
+        if hasattr(loop, "verbose"):
+            dumped_params["loop_verbose"] = loop.verbose
+
+    dumped_params["loop_state_dict"] = deepcopy(loop.state_dict())
+    return dumped_params
 
 
 def __scale_batch_reset_params(trainer: "pl.Trainer", steps_per_trial: int) -> None:
-    trainer.auto_scale_batch_size = None  # prevent recursion
-    trainer.auto_lr_find = False  # avoid lr find being called multiple times
-    trainer.fit_loop.max_steps = steps_per_trial  # take few steps
-    trainer.loggers = [DummyLogger()] if trainer.loggers else []
-    trainer.callbacks = []  # not needed before full run
-    trainer.limit_train_batches = 1.0
+    from pytorch_lightning.loggers.logger import DummyLogger
+
+    trainer.logger = DummyLogger() if trainer.logger is not None else None
+    trainer.callbacks = []
+
+    if trainer.state.fn == "fit":
+        trainer.limit_train_batches = 1.0
+        trainer.limit_val_batches = steps_per_trial
+        trainer.fit_loop.max_steps = steps_per_trial
+    else:
+        stage = trainer.state.stage
+        loop = getattr(trainer, f"{stage}_loop")
+        assert stage is not None
+        setattr(trainer, f"limit_{stage.dataloader_prefix}_batches", steps_per_trial)
+
+        if hasattr(loop, "verbose"):
+            loop.verbose = False
 
 
 def __scale_batch_restore_params(trainer: "pl.Trainer", params: Dict[str, Any]) -> None:
-    trainer.auto_scale_batch_size = params["auto_scale_batch_size"]
-    trainer.auto_lr_find = params["auto_lr_find"]
-    trainer.fit_loop.max_steps = params["max_steps"]
-    trainer.logger = params["logger"]
+    # TODO: There are more states that needs to be reset (#4512 and #4870)
```

Comment:
What would those be?

Reply:
one is profiler. Not sure about the other one. Added it here to investigate when this TODO is WIP